incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -150,6 +151,10 @@ func (r *HierarchyReconciler) reconcile(ctx context.Context, log logr.Logger, nm
 // fully written back to the apiserver yet, each namespace is reconciled in isolation (apart from
 // the in-memory forest) so this is fine.
 func (r *HierarchyReconciler) syncWithForest(log logr.Logger, nsInst *corev1.Namespace, inst *api.HierarchyConfiguration) {
+	if !ex[inst.ObjectMeta.Namespace] {
+		atomic.AddInt32(&mutexQ, 1)
```

Comment:
I think we found that there was no mutex contention, right? If so, I'd just delete these.